api/v1beta1/kafkacluster_types.go

```diff
@@ -377,6 +388,16 @@ func (c IngressServiceSettings) GetServiceType() corev1.ServiceType {
 	return c.ServiceType
 }
 
+// Replace %id in brokerHostnameTemplate with actual broker id
+func (c IngressServiceSettings) GetBrokerHostname(brokerId int32) string {
+	return strings.Replace(c.BrokerHostnameTemplate, "%id", strconv.Itoa(int(brokerId)), 1)
+}
+
+// We use -1 for ExternalStartingPort value to enable TLS on envoy
+func (c ExternalListenerConfig) TLSEnabled() bool {
```

Comment:
Determining whether TLS is enabled or not on an external listener based on the value of the `externalStartingPort` field IMHO is rather ambiguous than robust.

From the `KafkaCluster` CR it should be clear whether:
- TLS is enabled and provided by Kafka broker (currently this can be determined from the type of the external listener config - e.g.: ssl, sasl_ssl)
- TLS is enabled at the ingress controller (in this case whether TLS is enabled or not it should be determined from the specific ingress controller config)

Since having both TLS implemented at Kafka broker and ingress controller as well at the same time, doesn't seem to be a use case - at least for now - that should be supported there needs to be either:
- validation to not allow such config in KafkaCluster CR,
- or one of the two configs having priority over the other